### DISCUSSION TITLE
Tensorboard and WandB metrics log

### DISCUSSION THREAD

**Discussion by ramajoballester** [open-mmlab/mmdetection3d#2263] [Q&A] [answered]
```Branch: dev-1.x```.

Hi, I would like to ask how can I set the configs to use tensorboard and wandb to visualize my trainings. Tensorboard does not read the generated default logs, so I guess I'm missing something.

I'm kind of new to this project so I'm trying to get the grasp of it to the maximum detail possible. For instance, I'm training with default ```configs/pointpillars/pointpillars_hv_secfpn_8xb6-160e_kitti-3d-3class.py``` and for what I can see in the docs, I should include ```WandbVisBackend``` and ```TensorboardVisBackend``` somewhere in ```_base_/default_runtime.py```:

```
default_hooks = dict(
    timer=dict(type='IterTimerHook'),
    logger=dict(type='LoggerHook', interval=50),
    param_scheduler=dict(type='ParamSchedulerHook'),
    checkpoint=dict(type='CheckpointHook', interval=-1),
    sampler_seed=dict(type='DistSamplerSeedHook'),
    visualization=dict(type='Det3DVisualizationHook')
)
```

Am I right?

My question is how should I compose them in the configs and how can I see them with analyze_logs.py.

Thank you so much in advance.

**Comment by JingweiZhang12** [comment, marked as answer]
@ramajoballester Hi, thanks for your attention to our project. If you want to change the different visualizer backend, here:
https://github.com/open-mmlab/mmdetection3d/blob/dev-1.x/configs/_base_/datasets/kitti-3d-3class.py#L130
You can set: vis_backend = [dict(type='WandbVisBackend')] or vis_backend = [dict(type='TensorboardVisBackend')], or vis_backend = [dict(type='WandbVisBackend'), dict(type='WandbVisBackend')] and so on.
